src/toolkits/drawing_classifier/dc_data_iterator.hpp

```diff
@@ -73,6 +74,9 @@ class data_iterator {
   /** Defines the output of a data_iterator. */
   struct batch {
 
+    /* Number of examples in batch */
+    size_t num_samples;
```

Comment:
Let's give this a default value, so that a default-constructed instance of `batch` has a deterministic value

Reply:
same for data_iterator::param, add default value.